cupy/lib/polynomial.pyx

```diff
@@ -129,11 +129,20 @@ cdef class poly1d:
     def __pos__(self):
         return self
 
-    # TODO(Dahlia-Chehata): use polymul for non-scalars
     def __mul__(self, other):
+        if _should_use_rop(self, other):
+            return other.__rmul__(self)
         if cupy.isscalar(other):
+            # case: poly1d * python scalar
             return poly1d(self.coeffs * other)
```

Comment:
It is needed for the commented case

Reply:
Why? the input array will be trimmed in polymul routine.

Reply:
There is a difference in the result dtype in case of  poly1d * python scalar. `convolve ` in polymul uses `result_type` for both of its inputs. To work around this, the result need to be casted to numpy.float64 in case when python scalar is a rhs operand but casting can be bothersome if the original result is of complex type. Another solution is to handle this specifically for polymul function in the wrapper itself which you didn't favour before

Reply:
If so, you should write a comment about the background on this workaround.